module4/lessons/Week2/IntroToAPIs.md

```diff
@@ -0,0 +1,166 @@
+
+Clone down [this](https://github.com/turingschool-examples/CallingAPIsPractice) repository.
+
+The goal is not for you to understand all of the code, but to get practice calling APIs from an application and digging through the response object to access the data you want.
+
+You should only need to change the lines that start with `CHANGE THIS:` while completing the practice.
+
+```c#
+using System.Runtime.InteropServices.ComTypes;
+using System.Text.Json;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+class Program
+{
+    static async Task Main(string[] args)
+    {
+        // GitHub API endpoint
+        string apiUrl = "https://api.github.com/orgs/dotnet/repos"; //CHANGE THIS: THE URL
+
+        // Create an instance of HttpClient
+        using (HttpClient httpClient = new HttpClient())
+        {
+            // Set the User-Agent header required by GitHub API
+            // The header provides information about the client making the request and helps the server identify the type of client, its operating system, and version information.
+            // Postman sets this header automatically as something like PostmanRuntime/7.32.3
+            httpClient.DefaultRequestHeaders.UserAgent.ParseAdd("CallingAPIsPractice/1.0");
+
+            // Send the GET request to the API
+            HttpResponseMessage response = await httpClient.GetAsync(apiUrl);
+
+            // Read the response content as a string
+            string responseBody = await response.Content.ReadAsStringAsync();
+
+            // Use the Json library to turn the string into an array
+            JArray jsonResponse = JsonConvert.DeserializeObject<JArray>(responseBody);
+            Console.WriteLine(jsonResponse[0]["name"]); //CHANGE THIS: THE WAY YOU ACCESS THE RESPONSE DATA
+        }
+    }
+}
+
+```
+
+Takes notes on what URL and what C# code you used to output this information! We will talk through these as a group!
+
+<aside class="instructor-notes" markdown="1">
+    <p><strong>Instructor Note</strong><br>I would walk through the first one together. I would start by making the request in postman and then go bit by bit to get to the desired output. https://api.github.com/repos/dotnet/aspnetcore/pulls?sort=created </p>
+</aside>
+
+Questions:
+1. Output the username of the person who most recently created a pull request for the ASP.NET Core repository.
+1. Output the third person to fork the [MVC Movies](https://github.com/turingschool-examples/MvcMovieStarter) repository.
+1. What is the name of the repository in the .NET organization that is alphebetically last? What other information can you output about that repository formatted in the console as a useful summary?
+1. 🌶️ What are the usernames of the 20 most recently created pull request for the ASP.NET Core repository.
+1. 🌶️ What are the usernames of all of the users who forked the MVC Movies repository?
+

```

Comment:
It might be good to revisit the question `What clients have we used so far?` Just to make sure that students are making that connection.

Reply:
Good call, I'm going to modify the second CFU question to more specifically use the term client.